### PR TITLE
Alerting: Improve showing nextEvaluationDate for rules

### LINKED ISSUE #65125: Alerting: Confusing "next evaluation" on alert rule
**What happened**:

After the initial implementation in https://github.com/grafana/grafana/pull/64767 it seems there are 2 scenarios where computing the next occurrence is confusing:

1. We've not had a previous evaluation yet, so we're not computing the next one.
2. The previous occurrence we're using missing one or more ticks, meaning there has been a previous evaluation but has missed several theoretical evaluations because either the server restarted or the rule was updated before "evaluation group interval".

**What you expected to happen**:

We should figure out how to show these correctly, possibly we'd need a timestamp from the server that tells us at what time it started the evaluation ticker.

**How to reproduce it (as minimally and precisely as possible)**:

**Scenario 1**

1. Create a new alert rule in an evaluation group of "5m"
2. Observe that we don't have a "next evaluation"

**Scenario 2**

1. Create a new alert rule in an evaluation group of "1m"
2. Wait 1m for the first evaluation
3. Stop the server and wait _at least_ 1 minute
4. Observe the "next evaluation" to be one minute in the past since it's just computing "previous occurrence + evaluation interval"

We should also back-port this all the way to 9.3 since the initial PR was also back-ported that far.

### PR DESCRIPTION
**Pull request by VikaCep**
**What is this feature?**

In https://github.com/grafana/grafana/pull/64767 we added the **Next evaluation** column to the rules list view which shows the estimated time where the rule will be evaluated in the future. To calculate this value we use the `lastEvaluationDate` and group `interval` which we receive from the API and we add them up.
This PR improves that implementation taking into account of two extra scenarios:
- When the rule hasn't been evaluated yet and the column is empty
- When the rule has been evaluated in the past, missed some next evaluations due to several posible reasons like the server being down, and ending up in a `nextEvaluationDate` that is earlier than the current time. 

To address these two scenarios, when we don't have a defined/accurate `lastEvaluationDate` we make use of the group's `interval` value to show that the rule will be evaluated in a range between now + interval.

This way we reduce the confusion that might be caused by an empty "Next evaluation" column and also make sure not to display invalid past dates.

**Why do we need this feature?**

To address some specific scenarios when the rule's `lastEvaluationDate` is not accurate.

**Who is this feature for?**

All users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/65125

**Special notes for your reviewer:**

image

### CODE REVIEW

File: public/app/features/alerting/unified/components/rules/RulesTable.tsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import { css, cx } from '@emotion/css';
+import { isBefore } from 'date-fns';
+import formatDuration from 'date-fns/formatDuration';
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rules/RulesTable.tsx:3]: Any particular reason why we're not also destructuring this like the import above?

File: public/app/features/alerting/unified/components/rules/RulesTable.tsx [hunk 2]
```diff
@@ -116,21 +117,29 @@ function useColumns(showSummaryColumn: boolean, showGroupColumn: boolean, showNe
...
+    if (!isValidLastEvaluation || !isValidIntervalDuration || isGrafanaRulerRulePaused(rule)) {
       return;
     }
 
-    const lastEvaluationDate = Date.parse(rule.promRule?.lastEvaluation || '');
     const intervalDuration = parseDuration(rule.group.interval!);
+    const lastEvaluationDate = Date.parse(rule.promRule?.lastEvaluation || '');
     const nextEvaluationDate = addDurationToDate(lastEvaluationDate, intervalDuration);
+
+    //when `nextEvaluationDate` is a past date it means the `lastEvaluationDate` was inaccurate.
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rules/RulesTable.tsx:131]: I think it's more accurate (no pun intended) to say that the `lastEvaluation` was more than one evaluation interval ago.

**Review by gillesdemey — APPROVED** [review]
LGTM just a few nits :)

File: public/app/features/alerting/unified/components/rules/RulesTable.tsx (hunk 1 above)
Comment (VikaCep) [reply public/app/features/alerting/unified/components/rules/RulesTable.tsx]: No reason, fixed 👍

File: public/app/features/alerting/unified/components/rules/RulesTable.tsx (hunk 2 above)
Comment (VikaCep) [reply public/app/features/alerting/unified/components/rules/RulesTable.tsx]: Sounds better! changed it.

**Pull request merged**